tests/pytests/test_info_modules.py

```diff
@@ -187,13 +188,82 @@ def test_redis_info():
     conn.execute_command('DEL', f'h{i}')
 
   # Force-invoke the GC
-  forceInvokeGC(env, 'idx')
-
-  # Wait for GC to finish
-  time.sleep(2)
+  forceInvokeGC(env)
 
   # Call `INFO` and check that the data is updated accordingly
   res = env.cmd('INFO', 'MODULES')
   env.assertGreater(res['search_bytes_collected'], 0)
   env.assertGreater(res['search_total_cycles'], 0)
   env.assertGreater(res['search_total_ms_run'], 0)
+
+
+def test_counting_queries(env: Env):
```

Comment:
1. Consider adding timeout errors validations.
2. check that internal queries are not counted
3. MT environment?

Reply:
Another suggestion - in cluster mode test shards aggregation result